tests/factories.py

```diff
@@ -75,6 +78,62 @@ def create(
         )
 
 
+class RestrictionsFactory(Factory):
+    model = Restrictions
+    counter = 65
+
+    @classmethod
+    def create(
+        cls,
+        required=None,
+        min_occurs=None,
+        max_occurs=None,
+        min_exclusive=None,
+        min_inclusive=None,
+        min_length=None,
+        max_exclusive=None,
+        max_inclusive=None,
+        max_length=None,
+        total_digits=None,
+        fraction_digits=None,
+        length=None,
+        white_space=None,
+        pattern=None,
+        explicit_timezone=None,
+        nillable=None,
+    ):
+        return cls.model(
+            required=required,
+            min_occurs=min_occurs,
+            max_occurs=max_occurs,
+            min_exclusive=min_exclusive,
+            min_inclusive=min_inclusive,
+            min_length=min_length,
+            max_exclusive=max_exclusive,
+            max_inclusive=max_inclusive,
+            max_length=max_length,
+            total_digits=total_digits,
+            fraction_digits=fraction_digits,
+            length=length,
+            white_space=white_space,
+            pattern=pattern,
+            explicit_timezone=explicit_timezone,
+            nillable=nillable,
+        )
+
+
+class ExtensionFactory(Factory):
+    model = Extension
+    counter = 65
+
+    @classmethod
+    def create(cls, type=None, restrictions=None):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Redefining built-in 'type'](https://app.codacy.com/manual/tefra/xsdata/pullRequest?prid=5052528)